src/common/procedure/src/local/runner.rs

```diff
@@ -165,12 +165,21 @@ impl Runner {
             provider: self.manager_ctx.clone(),
         };
 
+        let mut exponential = self.exponential_builder.build();
         loop {
+            let mut retry_times = 0;
```

Comment:
Looks like we never reset the `exponential` when `execute_once()` succeeds.

suggestion: Could we wrap the retry logic into a method named `execute_once_with_retry()`? So we could keep this loop clear.
```rust
// We could use bool to represent whether to continue since we handle RetryLater and Failed in this method
fn execute_once_with_retry() -> Result<bool> {
    for dur in retry {
        match self.execute_once() {
            ...
        }
    }
}
```

You could refer to https://github.com/datafuselabs/opendal/blob/33346a8f857f8b75b81fd064644f57c04f340f84/src/layers/retry.rs#L583